samples/subsys/mgmt/mcumgr/smp_svr: Broken on board used in README

**Describe the bug**
Following instructions https://docs.zephyrproject.org/latest/samples/subsys/mgmt/mcumgr/smp_svr/README.html for sample samples/subsys/mgmt/mcumgr/smp_svr leads to compilation issue:

```
~/zephyrproject/zephyr/include/zephyr/device.h:85:41: error: '__device_dts_ord_DT_CHOSEN_zephyr_uart_mcumgr_ORD' undeclared here (not in a function)
```

**To Reproduce**
Follow sample instructions
See compilation issue

**Expected behavior**
Sample instructions should lead to an easy reproduction of the sample.

**Impact**
Confusing

--- a/boards/arm/frdm_k64f/frdm_k64f.dts
+++ b/boards/arm/frdm_k64f/frdm_k64f.dts
@@ -28,6 +28,7 @@
 		zephyr,shell-uart = &uart0;
 		zephyr,uart-pipe = &uart0;
 		zephyr,canbus = &flexcan0;
+		zephyr,uart-mcumgr = &uart0;
 	};
 
 	leds {

